File: examples/docs-code-snippets/src/test/java/examples/java/recipes/WebShopRecipeTest.java
package examples.java.recipes;

import com.ing.baker.compiler.RecipeCompiler;
import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class WebShopRecipeTest {

    @Test
    public void recipeShouldCompileWithoutValidationErrors() {
        var validationErrors = RecipeCompiler.compileRecipe(WebShopRecipe.recipe).getValidationErrors();
        assertTrue(
            String.format("Recipe compilation resulted in validation errors: \n%s", validationErrors),
            validationErrors.isEmpty()
        );
    }
}

File: examples/docs-code-snippets/src/test/kotlin/examples/kotlin/recipes/WebShopRecipeTest.kt
package examples.kotlin.recipes

import com.ing.baker.compiler.RecipeCompiler
import com.ing.baker.recipe.kotlindsl.ExperimentalDsl
import org.junit.Test

@ExperimentalDsl
class WebShopRecipeTest {
    @Test
    fun `recipe should compile without validation errors`() {
        val validationErrors = RecipeCompiler.compileRecipe(WebShopRecipe.recipe).validationErrors
        assert(validationErrors.isEmpty()) {
            "Recipe compilation resulted in validation errors: \n${validationErrors.joinToString(separator = "\n")}"
        }
    }
}